sound/soc/sof/sof-audio.c

```diff
@@ -162,7 +169,7 @@ int sof_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 	if (++swidget->use_count > 1)
 		return 0;
 
-	ret = sof_pipeline_core_enable(sdev, swidget);
+	ret = snd_sof_dsp_core_get(sdev, swidget->core);
```

Comment:
it feels like we are blurring layers.
when you enable a pipeline, you need to internally increase the refcount for the core used by that pipeline.
I don't see why we would directly manipulate the core refcount.

Reply:
> it feels like we are blurring layers.
> when you enable a pipeline, you need to internally increase the refcount for the core used by that pipeline.
> I don't see why we would directly manipulate the core refcount.

We don't need to treat pipeline as exception, the scheduler widget will represent the pipeline and since it is a swidget also, it will go with the unified snd_sof_dsp_core_get() interface and the sof_pipeline_core_enable() is no needed anymore.

Reply:
According to current IPC and feature IPC4, core usage is not defined by pipeline, just by its components or modules which may run on different dsp cores, so we should let their widgets to set power status.

Reply:
@keyonjie concerning pipeline special treatment: what about the order? Can it happen, that, say, the pipeline is assigned to core 1, but the first widget / component that we send to the firmware is on a different (0 or 2) core. So, we don't yet power up 1, but the firmware identifies, that the pipeline runs on core 1 so it tries to use it for some pipeline-level initialisation?

Reply:
> According to feature IPC4, core usage is not defined by pipeline, just by its components or modules which may run on different dsp cores, so we should let their widgets to set power status.

Thanks @RanderWang, that is a very important point. At this point, it's clear we need a timeout and first agree on requirements for multi-core, then review a 2-page max design document before we jump into code reviews. This is not productive if we don't even know what we want to implement. @keyonjie @ranj063 @kv2019i any objections?

Reply:
@plbossart I agree. With the requirements being different for IPC4, we absolutely need to pause and work out the details before we jump into implementation.